### PR TITLE
feat(publish): Add optional post-release script

### PR DESCRIPTION
**Pull request by BYK**
This PR adds an optional post-release script which is needed for
projects like [getsentry/sentry](https://git.io/JIs4n) and
[getsentry/onpremise](https://git.io/JIs4l). This would help keeping
the common publish/release repo and its downstream project independent.

### CODE REVIEW

File: src/commands/publish.ts
Comment (tonyo) [inline src/commands/publish.ts:422]:
```suggestion
    logger.warn('Not running the post-release command: no command specified');
```

File: src/commands/publish.ts
Comment (tonyo) [inline src/commands/publish.ts:431]: Would suggest to make these this message and the one above `info` (or lower) instead of `warn`, most of the people will not want to see a warning about a thing they don't really care about

**Review by tonyo — APPROVED** [review]

**Review by chadwhitacre — APPROVED** [review]
I like the feature. It makes sense to support the specific use case of bumping the version post-release, and in general as a parallel to the pre-release hook. That said, this PR has the speed/quality dial set pretty far on "speed." I'm approving in case "speed" is where we _want_ the dial set. If we want to turn the dial more towards quality and take on less tech debt, here are some suggestions:

1. Refactor to deduplicate code between `runPreReleaseCommand` and `runPostReleaseCommand`.
1. Test the post-release command hook in the test suite. There are tests for the pre-release hook. Either duplicate or refactor, based on decision under (1).
1. Update the documentation in the README. There's a section for the pre-release script; either duplicate or refactor.

**Comment by BYK** [comment]
@chadwhitacre will address points 2 and 3 in your feedback before merging, thanks a lot! (I may take a stab at 1 but no promises).

**Review by jan-auer — APPROVED** [review]

**Comment by BYK** [comment]
I need another review, especially for the README changes.

@chadwhitacre - I've opted not to refactor/DRY the common parts in script runners as the optional nature of the `post-release` script makes this effort mostly fruitless (adds more complexity).

File: README.md
Comment (chadwhitacre) [inline README.md:258]:
```suggestion
[Post-release script conventions section](#post-release-script-conventions)
```

File: README.md
Comment (chadwhitacre) [inline README.md:910]:
```suggestion
by specifying the `preReleaseCommand` configuration option in `craft.yml`.
```

**Review by chadwhitacre — APPROVED** [review]
Couple doc nits, otherwise lgtm.

File: README.md
Comment (tonyo) [inline README.md:262]: -> `postReleaseCommand`

File: README.md
Comment (tonyo) [inline README.md:940]: -> `./scripts/post-release.sh`

**Review by tonyo — COMMENTED** [review]
A couple small things

File: README.md
Comment (BYK) [reply README.md:940]: #146 - sorry about this 😞

**Pull request merged**
